### PR TITLE
[#7] Implement `MonadJwt` effect

### LINKED ISSUE #7: Implement `MonadJwt` effect
- [ ] Jwt data types and functions
- [ ] `MonadJwt` effect
- [ ] Roundtrip tests

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #7

Needs to be improved (please, send help):

1. Better naming for `ClaimsMap <-> JwtPayload` functions, Please, send help!
2. Later: figure out how to encode/decode different types of `JwtPayload`s. Maybe after #16

### CODE REVIEW

**Comment by chshersh** [comment]
@vrom911 This is ready for review now :+1:

File: src/CakeSlayer/Jwt.hs [hunk 1]
```diff
@@ -0,0 +1,180 @@
+{- | This module provides convenience wrappers around JWT provided by @jwt@
```
Comment (vrom911) [inline src/CakeSlayer/Jwt.hs]:
```suggestion
{- | This module provides convenient wrappers around JWT provided by @jwt@
```

File: test/Spec.hs [hunk 2]
```diff
@@ -3,16 +3,24 @@ module Main (main) where
...
+import Test.Gen (genInt, genText)
+import Test.Jwt (jwtRoundtrip)
 import Test.Password (pwdHashVerify)
 
 
 hedgehogTests :: Group
 hedgehogTests = Group "Roundtrip properties"
-    [ pwdHashVerify `named` "verify . hash ≡ True"
+    [ "verify  . hash        ≡ True" `named` pwdHashVerify
+    , "fromMap . toMap @Int  ≡ Just" `named`
```
Comment (vrom911) [inline test/Spec.hs]: It's not clear from the description that these are JWT related tests

File: src/CakeSlayer/Jwt.hs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+-- Secret
+----------------------------------------------------------------------------
+
+{- | This monad represents effect to create and verify JWT.
+
+TODO: parametrize 'JwtPayload' when we figure out how to do this
+-}
+class Monad m => MonadJwt m where
+    mkJwtToken
+        :: Int              -- ^ Token expiry in seconds
```
Comment (arbus) [inline src/CakeSlayer/Jwt.hs]: Should we add a TODO here to convert this to the `Seconds` newtype once it is in?

Comment (chshersh) [reply src/CakeSlayer/Jwt.hs]: I've created an issue about seconds:

* https://github.com/Holmusk/cake-slayer/issues/20

File: test/Spec.hs (hunk 2 above)
Comment (chshersh) [reply test/Spec.hs]: Done!

**Pull request merged**